core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -91,7 +94,11 @@ private[spark] class EventLoggingListener(
   private[scheduler] val loggedEvents = new ArrayBuffer[JValue]
 
   // Visible for tests only.
-  private[scheduler] val logPath = getLogPath(logBaseDir, appId, appAttemptId, compressionCodecName)
+  private[scheduler] val logPath = getLogPath(
+    logBaseDir, appId, appAttemptId, compressionCodecName)
+
+  private val latestMetrics = new HashMap[String, SparkListenerExecutorMetricsUpdate]
+  private val modifiedMetrics = new HashMap[String, SparkListenerExecutorMetricsUpdate]
```

Comment:
can you indicate that they are keyed by executorId, eg., rename to `executorIdToLatestMetrics`